### PR TITLE
Inserter: Should receive focus on open

### PR DESCRIPTION
**Pull request by jeryj**
See https://github.com/WordPress/gutenberg/issues/67755
## What?

We were relying on the tabs component to set an initial tab for us, and then focusing it afterwards. It seems like something has changed within the tabs component's behavior where the selected tab is getting set too late, and our auto-focus check has already completed. Passing which tab we want selected fixes it.

## Why?

Consistent behavior and readability. Since we will always have an initial tab selected, it's easier to understand the code if there's always a default tab set instead of having the tabs component guess which one for us. 

## How?

Sets an initial tab to be focused if none is manually passed.

## Testing Instructions

- Click the inserter toggle button
- There should be no visible focus outline
- Press the right arrow key
- Focus should be on the pattern tab

### Testing Instructions for Keyboard

- Move focus to the inserter toggle button
- Press enter
- The blocks tab should be focused with a visible outline

## Screenshots or screencast 

|Before|After|
|-|-|
|||

### CODE REVIEW

File: test/e2e/specs/site-editor/site-editor-inserter.spec.js [hunk 1]
```diff
@@ -28,32 +28,51 @@ test.describe( 'Site Editor Inserter', () => {
...
+			await InserterUtils.blockLibrary
+				.getByRole( 'option', { name: 'Buttons' } )
+				.click();
+
+			await expect
+				.poll( editor.getBlocks )
+				.toMatchObject( [ ...beforeBlocks, { name: 'core/buttons' } ] );
+
+			await InserterUtils.closeBlockLibrary();
+		} );
```
Comment (jeryj) [inline test/e2e/specs/site-editor/site-editor-inserter.spec.js:75]: Moving this into a step for better organization and saving a bit of time on e2e test runs.

File: packages/block-editor/src/components/inserter/menu.js [hunk 2]
```diff
@@ -82,6 +82,8 @@ function InserterMenu(
 		if ( isZoomOutMode ) {
 			return 'patterns';
 		}
+
+		return 'blocks';
```
Comment (Mamaduka) [inline packages/block-editor/src/components/inserter/menu.js:86]: Not sure why we missed this. The `getInitialTab` should always return value.

Comment (jeryj) [reply packages/block-editor/src/components/inserter/menu.js:86]: Yeah, I'm surprised we missed it too. I guess we know we're human :)

**Review by Mamaduka — APPROVED** [review]
Thanks, @jeryj! This works as expected ✅ 

I've unlinked the issue since the error remains.

## Screenshot

https://github.com/user-attachments/assets/9f1eb457-717c-4f31-ba94-78b691ee1c50

**Comment by youknowriad** [comment]
This impacted the "inserter open" metric. That said, it might just because previously we were just rendering nothing right?

**Comment by jeryj** [comment]
@youknowriad That's very odd. Since this impacted the inserter open metric, I'd say there's a different underlying issue this is exposing. It doesn't change what is rendered, just tells the inserter what tab to move focus to when it opens.

**Comment by Mamaduka** [comment]
Yeah, a 20%+ increase is odd, but then the previous version of `getInitialTab` was incorrect. We should probably compare profiles before and after this got merged.

**Comment by jeryj** [comment]
Previously, the `selectedTab` was set as `undefined` on the first pass, and then updated via the `Tabs` component `blocks`. From looking at the performance test, it checks the toggle button to see [when it gets an `expanded` state](https://github.com/WordPress/gutenberg/blob/trunk/test/performance/specs/post-editor.spec.js#L390). 

Maybe with `undefined` for the `selectedTab` on the first render, there was no tab content rendered, so the toggle button was able to get the `expanded` state set without rendering the block tab, so it was faster?

**Comment by youknowriad** [comment]
> Maybe with undefined for the selectedTab on the first render, there was no tab content rendered, so the toggle button was able to get the expanded state set without rendering the block tab, so it was faster?

Yes, this is my guess, in which case, there's nothing to revert, but maybe we can try to make the metric independent of the "sync/asynchronous" behavior and count the time until first block is rendered or something.

**Pull request merged**